lib/Models/createCatalogItemFromFileOrUrl.js

```diff
@@ -30,47 +32,68 @@ var createCatalogItemFromFileOrUrl = function(terria, fileOrUrl, dataType, confi
-Click OK to upload the file to the '+terria.appName+' conversion service now.  Or, click Cancel \
-and the file will not be uploaded or added to the map.')) {
-            return when.reject('The user declined to use the conversion service.');
+        var newCatalogItem = createCatalogItemFromUrl(name, terria);
+        if (newCatalogItem.type === 'ogr') {
+            if (name.match(/\.(shp|jpg|jpeg|pdf|xlsx|xls|tif|tiff|png|txt|doc|docx|xml|json)$/)) {
+                terria.error.raiseEvent(new TerriaError({
+                    title: 'Unsupported file type',
+                    message: 'This file format is not supported by ' + terria.appName + '. Directly supported file formats include: ' + 
+                    '<ul><li>.geojson</li><li>.kml, .kmz</li><li>.csv (in <a href="https://github.com/NICTA/nationalmap/wiki/csv-geo-au">csv-geo-au format</a>)</li></ul>' + 
+                    'File formats supported through the online conversion service include: ' +
+                    '<ul><li>Shapefile (.zip)</li><li>MapInfo TAB (.zip)</li><li>Possibly other <a href="http://www.gdal.org/ogr_formats.html">OGR Vector Formats</a></li></ul>'
+                }));
+                return undefined;
+            } 
+            return getConfirmation(confirmConversion, 'This file is not directly supported by '+terria.appName+'.\n\n' +
+                'Do you want to try uploading it to the '+terria.appName+' conversion service? This may work for ' +
+                'small, zipped Esri Shapefiles or MapInfo TAB files.')
+                .then(function() { return loadItem(newCatalogItem, name, fileOrUrl); });
+        } else {
+            return loadItem(newCatalogItem, name, fileOrUrl); // It's a file we support directly
         }
-
-        newCatalogItem = createCatalogMemberFromType('ogr', terria);
+    } else if (dataType === 'other') { // user explicitly chose "Other (use conversion service)"
+        return getConfirmation(confirmConversion, 'Ready to upload your file to the ' + terria.appName + ' conversion service?')
+            .then(function() { 
+                return loadItem(createCatalogMemberFromType('ogr', terria), name, fileOrUrl); 
+            });
     } else {
-        newCatalogItem = createCatalogMemberFromType(dataType, terria);
+        return loadItem(createCatalogMemberFromType(dataType, terria), name, fileOrUrl);
     }
+};
 
-    var lastSlashIndex = name.lastIndexOf('/');
+function getConfirmation(confirmConversion, message) {
+    if (!confirmConversion)
+        return when(true);
+    
+    var d = when.defer(); // there's no `when.promise(resolver)` in when 1.7.1
+    PopupMessageConfirmationViewModel.open('ui', { 
+        confirmText: 'Upload', 
+        title: 'Use conversion service?',
+        message: message,
+        confirmAction: d.resolve,
+        enableDeny: true,
+        denyAction: function() { this.close(); } // we don't call resolve, so just leave the promise hanging?
```

Comment:
Good question (when the popup is cancelled, "we don't call resolve so just leave the promise hanging?") - can I refer this to @kring?